plugins/woocommerce/includes/admin/views/html-admin-page-product-export.php

```diff
@@ -12,16 +12,68 @@
 wp_enqueue_script( 'wc-product-export' );
 
 $exporter = new WC_Product_CSV_Exporter();
+
+$product_ids_to_export = array();
+$is_exporting_specific = false;
+
+if ( ! empty( $_GET['product_ids'] ) ) {
+	check_admin_referer( 'export-selected-products' );
+
+	$ids_raw               = explode( ',', sanitize_text_field( wp_unslash( $_GET['product_ids'] ) ) );
+	$product_ids_to_export = array_filter( array_map( 'absint', $ids_raw ) );
+
+	if ( ! empty( $product_ids_to_export ) ) {
+		$is_exporting_specific = true;
+	}
+}
 ?>
+
 <div class="wrap woocommerce">
 	<h1><?php esc_html_e( 'Export Products', 'woocommerce' ); ?></h1>
 
+	<?php
+	if ( $is_exporting_specific ) {
+		$clear_url = remove_query_arg( 'product_ids' );
+		$count     = count( $product_ids_to_export );
+		$notice    = sprintf(
+			// translators: %1$d: Number of products, %2$s: URL to clear selection.
+			_n(
+				'You are exporting the %1$d product you selected in the previous step. <a href="%2$s">Clear selection</a> to export all products.',
```

Comment:
Pinging @jimjasson for a copy review here. At least the singular one could be 
```
You are exporting the product you selected in the previous step.
-or-
You are about to export one product. Clear selection to export all products.
```

Jason, suggestions? :)

Reply:
> You are about to export 1 product. To export all products, clear your selection.